app/controllers/concerns/upload_validations.rb

```diff
@@ -0,0 +1,35 @@
+module UploadValidations
+  extend ActiveSupport::Concern
+
+  private
+
+  def validate_file_upload!
```

Comment:
Whilst this could work, I am not keen on defining additional models and bringing in extra gems to the codebase where we can avoid it - we only have a few validation rules so I think this would be overkill.